R/addins.R

```diff
@@ -13,103 +13,15 @@ style_active_file <- function() {
 #' one thing: You can highlight also just parts of lines.
 #' @importFrom rlang seq2
 style_active_region <- function() {
-  context <- find_active_context()
-  if (all(context$start == context$end)) stop("No region selected")
-  all_text <- utf8::read_lines_enc(context$path)
-  styled_expr <- style_region(all_text, context)
-  neighbourhood <- extract_neighbourhood(all_text, context)
-
-  styled_lines <- complete_styled_expr(context, styled_expr, neighbourhood)
-
-  merged_text <- append(neighbourhood$lines_not_to_style, styled_lines, context$start[1] - 1)
-  utf8::write_lines_enc(merged_text, context$path)
-}
-
-#' Helper to extract and preprocess relevant attributes from
-#' [rstudioapi::getActiveDocumentContext()].
-#'
-find_active_context <- function() {
   context <- get_rstudio_context()
-  path <- context$path
-  start <- context$selection[[1]]$range$start
-  end <- context$selection[[1]]$range$end
-  if (end[2] == 1 & !all(start == end)) {
-    end[1] <- end[1] - 1
-    end[2] <- 1000000L # because of range constraint in substr()
-  }
-  list(start = start, end = end, path = path)
+  text <- context$selection[[1]]$text
+  if (all(nchar(text) == 0)) stop("No code selected")
+  out <- style_text(text)
+  rstudioapi::modifyRange(
```

Comment:
Can we extract a wrapper around `modifyRange()` and mock that wrapper?

Reply:
The problem is that `rstudioapi` assigns the document ID dynamically, and `modifyRange()` needs this ID to select the document to modify, so I think we can't mock. The approach would have been something like 3fac3a9.
I suggest we just leave it without testing for now.

Reply:
The problem is that when modifying the file with `modifyRange()` we need the document id of the file to modify, which we get with `getActiveDocumentContext()` when run interactively. However, I don't see a way to get id without running `getActiveDocumentContext()` and ids are assigned dynamically (which can be tested by simply opening a file and tet the active context of it, closing it and get the context again).